_posts/2019-08-26-how-to-use-gruntwork-infrastructure-as-code-library.adoc

```diff
@@ -1399,6 +1409,272 @@ module "vpc" {
+
+1. Next, add an SSH key to the new GitHub machine user. GitHub has https://help.github.com/en/enterprise/2.19/user/github/authenticating-to-github/connecting-to-github-with-ssh[easy-to-follow instructions].
+
+1. Now, add the private portion of the SSH key to TFC. You'll find the option under SSH Keys in the TFC organization settings. We called ours _Gruntwork access_.
+
+1. Set up a new workspace and link it to your code repository that uses Gruntwork's module as a source.
+
+1. In the Workspace settings, in the SSH key section, you can now choose the Gruntwork access key.
+
+.Configuring a TFC Workspace to access Gruntwork repositories
+image::/assets/img/guides/infrastructure-as-code-library/tfc_ssh_key.png[]
+
+
+When TFC runs your module, it will use the configured SSH key to clone the Gruntwork repository during the run, and will continue with creating the SNS topic you defined. You can reuse this SSH key across any other workspaces that depend on Gruntwork modules.
+
+That's it! You can use this pattern for any of Gruntwork's Terraform modules. This even works for modules like github.com/gruntwork-io/terraform-aws-eks[`terraform-aws-eks`] which need to download external dependencies (like https://github.com/gruntwork-io/kubergrunt[`kubergrunt`]). Those modules use the special https://github.com/gruntwork-io/package-terraform-utilities/tree/master/modules/executable-dependency[`executable-dependency`] module to install the external dependency within the executor at run time.
+
+In the future, Gruntwork may offer modules through  a private Terraform registry to further simplify the process.
+
+[[terragrunt_tfc_tfe]]
+==== Using Terragrunt with TFC and TFE
+Configuring Terragrunt to work with TFC/TFE is a bit more involved. As mentioned previously, TFC and TFE execute
+`terraform` commands under the hood, and the executors that run these commands aren't able to execute the Terragrunt
+binary. Running Terragrunt directly through the TFC UI isn't an option.
+
+However, Terragrunt can rely on TFC's https://www.terraform.io/docs/cloud/run/cli.html#implicit-workspace-creation[implicit workspace creation] feature to automatically create workspaces when Terragrunt runs. This feature allows you to use Terragrunt to invoke TFC runs from your local CLI. Once you're up and running, the state will be maintained in TFC like any other workspace, and the standard plan and apply workflow will function normally. There are several steps involved to make this work:
+
+1. Generate a `backend` block to use TFC/TFE for remote state storage and execution.
+1. Generate a `terraform.auto.tfvars` for submitting variables.
+1. Configure authentication to providers.
+
+In the sections below we will cover each step in turn.
+
+===== Setting up
+
+Consider the following Terragrunt directory structure:
+
+[source]
+----
+.
```

Comment:
NIT: give the folder a name, such as `infrastructure-live`. Then in the `terragrunt.hcl` code snippets below, you can specify the full path (i.e., `infrastructure-live/terragrunt.hcl` and `infrastructure-live/dev/us-east-1/sns/terragrunt.hcl`) so it's easier for the reader to tell which file they are looking at.